packages/@aws-cdk/aws-ecs/README.md

```diff
@@ -1,2 +1,142 @@
+
+To create a cluster, go:
+
+```ts
+const cluster = new ecs.FargateCluster(this, 'Cluster', {
+  vpc: vpc
+});
+```
+
+## TaskDefinition
+
+A `TaskDefinition` describes what a single copy of a **Task** should look like.
+A task definition has one or more containers; typically, it has one
+main container (the *default container* is the first one that's added
+to the task definition, and it will be marked *essential*) and optionally
+some supporting containers which are used to support the main container,
+doings things like upload logs or metrics to monitoring services.
+
+To add containers to a `TaskDefinition`, call `addContainer()`:
+
+```ts
+const taskDefinition = new ecs.FargateTaskDefinition(this, 'TaskDef', {
+  memoryMiB: '512'
+  cpu: 256
+});
+
+taskDefinition.addContainer('main', {
+  // Use an image from DockerHub
+  image: ecs.DockerHub.image('amazon/amazon-ecs-sample')
+});
+```
+
+### Images
+
+Images supply the software that runs inside the container. Images can be
+obtained from either DockerHub or from ECR repositories:
+
+* `ecs.DockerHub.image(imageName)`: use a publicly available image from
+  DockerHub.
+* `repository.getImage(tag)`: use the given ECR repository as the image
+  to start.
+
+## Service
+
+A `Service` instantiates a `TaskDefinition` on a `Cluster` a given number of
+times, optionally associating them with a load balancer. Tasks that fail will
+automatically be restarted.
+
+```ts
+const taskDefinition;
+
+const service = new ecs.Ec2Service(this, 'Service', {
+  cluster,
+  taskDefinition,
+  desiredCount: 5
+});
+```
+
+### Include a load balancer
+
+`Services` are load balancing targets and can be directly attached to load
+balancers:
+
+```ts
+const service = new ecs.FargateService(this, 'Service', { /* ... */ });
+
+const lb = new elbv2.ApplicationLoadBalancer(this, 'LB', { vpc, internetFacing: true });
+const listener = lb.addListener('Listener', { port: 80 });
+listener.addTargets('ECS', {
+  port: 80,
+  targets: [service]
+});
+```
+
+There are two higher-level constructs available which include a load balancer for you:
+
+* `LoadBalancedFargateService`
+* `LoadBalancedEc2Service`
+
+## Task AutoScaling
+
+You can configure the task count of a service to match demand. Task AutoScaling is
+configured by calling `autoScaleTaskCount()`:
+
+```ts
+const scaling = service.autoScaleTaskCount({ maxCapacity: 10 });
+scaling.scaleOnCpuUtilization('CpuScaling', {
+  targetUtilizationPercent: 50
+});
+```
+
+Task AutoScaling is powered by *Application AutoScaling*. Refer to that for
+more information.
+
+## Instance AutoScaling
+
+If you're running on Fargate, AWS will manage the physical machines that your
+containers are running on for you. If you're running an ECS cluster however,
+your EC2 instances might fill up as your number of Tasks goes up.
+
+To avoid placement errors, you will want to configure AutoScaling for your
+EC2 instance group so that your instance count scales with demand.
+
+TO BE IMPLEMENTED
```

Comment:
Add a `Roadmap` section at the end and describe things you are planning instead of `TO BE IMPLEMENTED`